app/slashcommands/command_custom_status.go

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package slashcommands
+
+import (
+	"strings"
+
+	goi18n "github.com/mattermost/go-i18n/i18n"
+
+	"github.com/mattermost/mattermost-server/v5/app"
+	"github.com/mattermost/mattermost-server/v5/mlog"
+	"github.com/mattermost/mattermost-server/v5/model"
+)
+
+type CustomStatusProvider struct {
+}
+
+const (
+	CmdCustomStatus      = app.CmdCustomStatusTrigger
+	CmdCustomStatusClear = "clear"
+
+	DefaultCustomStatusEmoji = "speech_balloon"
+)
+
+func init() {
+	app.RegisterCommandProvider(&CustomStatusProvider{})
+}
+
+func (*CustomStatusProvider) GetTrigger() string {
+	return CmdCustomStatus
+}
+
+func (*CustomStatusProvider) GetCommand(a *app.App, T goi18n.TranslateFunc) *model.Command {
+	return &model.Command{
+		Trigger:          CmdCustomStatus,
+		AutoComplete:     true,
+		AutoCompleteDesc: T("api.command_custom_status.desc"),
+		AutoCompleteHint: T("api.command_custom_status.hint"),
+		DisplayName:      T("api.command_custom_status.name"),
+	}
+}
+
+func (*CustomStatusProvider) DoCommand(a *app.App, args *model.CommandArgs, message string) *model.CommandResponse {
+	if !a.Config().FeatureFlags.CustomUserStatuses || !*a.Config().TeamSettings.EnableCustomUserStatuses {
+		return &model.CommandResponse{}
```

Comment:
Maybe a more explanatory error about why is not doing nothing? //cc @esethna

Reply:
@esethna This is the case when the custom status setting is off and the `/status` slash command is executed.

Although we don't show any autocomplete text, the user can still execute the command. We don't take any action in this case. It probably makes sense to show a helpful error message instead.

What should be the slash command error message in this case?

P.S. probably not a blocker for this PR.

Reply:
I think it should be the standard error message for an unrecognized slash command:
 
![image](https://user-images.githubusercontent.com/14333569/108122484-33e55580-7059-11eb-976f-d8eabf4a2bdc.png)

Reply:
@esethna This is done with the latest commit.

![image](https://user-images.githubusercontent.com/35728906/108123598-21f0ca80-70cc-11eb-8f61-3109fdd93283.png)